app/controllers/checkouts_controller.rb

```diff
@@ -0,0 +1,95 @@
+class CheckoutsController < ApplicationController
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized, except: :update
+  protect_from_forgery except: :update
+  before_action :store_location
+
+  def checkout
+    authorize :checkout
+  end
+
+  def process_payment
+    authorize :checkout
+
+    # get payment method, if blank, it's credit card, if not, it's bank_slip
+    param          = (params[:token].empty? ? "method" : "token").to_sym
+    payment_method = params[:method] == '' ? 'credit_card' : 'bank_slip'
+
+    # initialize the purchase object
+    purchase = current_user.purchases.create
+    # Add all current items from cart to orders and purchase
+    session[:shopping_cart].each do |item|
+      offer = Offer.find(item['id'])
+      order = Order.create(offer: offer,
+                           purchase: purchase,
+                           offer_value: offer.value,
+                           quantity: item['quantity'])
+    end
+
+    charge = Iugu::Charge.create({
+      # here is where the method is determined, :token for CC or :method bank_slip
+      param => params[param],
+      email: current_user.email,
```

Comment:
Yes, this can be changed, but for now I'll let the user just fill the informations about the card, once that we don't need to confirm CPF or Payer name to make the purchase.